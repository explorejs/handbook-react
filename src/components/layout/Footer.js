import React from "react";

const Footer = ({ theme }) => (
  <footer
    style={{
      justifyContent: "center",
      backgroundColor: theme["background-color"],
      color: theme.color,
      display: "flex",
      padding: "1rem 0",
      width: "100vw",
    }}
  >
    <p
      style={{
        color: theme.color,
        backgroundColor: theme.backgroundColor,
        fontFamily: "Lato",
        fontSize: "1rem",
      }}
    >
      <span> ExploreJS </span>Â© {new Date().getFullYear()}
    </p>
  </footer>
);

export default Footer;
