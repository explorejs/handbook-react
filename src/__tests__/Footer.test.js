import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Footer from "../components/layout/Footer";

import THEME from "../theme.json";

describe("test Footer", () => {
  test("correct year & explore js in footer", () => {
    const { getByText } = render(<Footer theme={THEME["light"]} />);
    expect(getByText("ExploreJS")).toBeTruthy();
    expect(getByText("Â© 2021")).toBeTruthy();
  });

  it("renders correctly", () => {
    const result = renderer.create(<Footer theme={THEME["light"]} />).toJSON();
    expect(result).toMatchSnapshot();
  });
});
