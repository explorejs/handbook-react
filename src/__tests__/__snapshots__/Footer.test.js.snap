// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Footer renders correctly 1`] = `
<footer
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
      "display": "flex",
      "justifyContent": "center",
      "padding": "1rem 0",
      "width": "100vw",
    }
  }
>
  <p
    style={
      Object {
        "backgroundColor": undefined,
        "color": "black",
        "fontFamily": "Lato",
        "fontSize": "1rem",
      }
    }
  >
    <span>
       ExploreJS 
    </span>
    Â© 
    2022
  </p>
</footer>
`;
